File: models/payroll.js
const mongoose = require('mongoose');

const payrollSchema = new mongoose.Schema({
    // ربط السجل بالشركة لضمان فصل البيانات
    companyId: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'Company', 
        required: [true, 'يجب تحديد الشركة'] 
    },
    // ربط السجل بالموظف
    employeeId: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'Employee', 
        required: [true, 'يجب تحديد الموظف'] 
    },
    // الشهر والسنة (مثال: "2026-04")
    month: { 
        type: String, 
        required: [true, 'يجب تحديد الشهر والسنة'],
        trim: true 
    },
    /**
     * تفاصيل الحسبة كاملة (Snapshot)
     * بنخزن هنا الأرقام النهائية وقت الحسبة عشان لو مرتب الموظف اتغير مستقبلاً 
     * يفضل السجل القديم محتفظ بأرقامه صح
     */
    payload: {
        grossSalary: { type: Number, required: true },
        netSalary: { type: Number, required: true },
        taxAmount: { type: Number, default: 0 },
        insuranceEmployee: { type: Number, default: 0 }, // حصة الموظف
        insuranceCompany: { type: Number, default: 0 },  // حصة الشركة
        totalAdditions: { type: Number, default: 0 },
        totalDeductions: { type: Number, default: 0 },
        details: { type: Object } // تفاصيل إضافية مرنة
    },
    // حالة الدفع (مدفوع، معلق، مراجعة)
    status: {
        type: String,
        enum: ['Pending', 'Approved', 'Paid'],
        default: 'Pending'
    },
    createdAt: { 
        type: Date, 
        default: Date.now 
    }
});

/**
 * تحسين الأداء للتقارير (Indexing)
 */
// لمنع تكرار عمل كشف مرتب لنفس الموظف في نفس الشهر مرتين
payrollSchema.index({ employeeId: 1, month: 1 }, { unique: true });
// لسرعة جلب مرتبات شهر معين لشركة معينة
payrollSchema.index({ companyId: 1, month: 1 });

/**
 * التصدير النهائي (Serverless & Vercel Optimized)
 */
module.exports = mongoose.models.Payroll || mongoose.model('Payroll', payrollSchema);

File: models/payroll.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// Payroll يمثل سجل الراتب الشهري لموظف معين (Snapshot)
type Payroll struct {
	// التعديل: خليناه _id في الـ json عشان الـ Frontend يعرف يقرأ الـ ID من MongoDB
	ID         primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
	CompanyID  primitive.ObjectID `bson:"companyId" json:"companyId"`
	EmployeeID primitive.ObjectID `bson:"employeeId" json:"employeeId"`
	
	// تنسيق الشهر "YYYY-MM" (مثال: 2026-02) مهم جداً للتسلسل Sequential
	Month      string             `bson:"month" json:"month"` 
	
	Payload    PayrollPayload     `bson:"payload" json:"payload"`
	Status     string             `bson:"status" json:"status"` // Pending, Approved, Paid
	CreatedAt  time.Time          `bson:"createdAt" json:"createdAt"`
}

// PayrollAdjustment لحفظ تفاصيل كل إضافة أو خصم
type PayrollAdjustment struct {
	Name       string  `bson:"name" json:"name"`
	Amount     float64 `bson:"amount" json:"amount"`
	IsExempted bool    `bson:"isExempted" json:"isExempted"` 
}

// PayrollPayload تفاصيل الحسبة المالية الكاملة
type PayrollPayload struct {
	FullBasic              float64 `bson:"fullBasic" json:"fullBasic"`
	FullTransportation     float64 `bson:"fullTransportation" json:"fullTransportation"`
	WorkingDays            int     `bson:"workingDays" json:"workingDays"`
	ProratedBasic          float64 `bson:"proratedBasic" json:"proratedBasic"`
	ProratedTransportation float64 `bson:"proratedTransportation" json:"proratedTransportation"`
	
	AdditionsList       []PayrollAdjustment `bson:"additionsList" json:"additionsList"`
	DeductionsList      []PayrollAdjustment `bson:"deductionsList" json:"deductionsList"`
	TotalAdditions      float64             `bson:"totalAdditions" json:"totalAdditions"`
	TotalDeductions     float64             `bson:"totalDeductions" json:"totalDeductions"`

	InsuranceSalary     float64 `bson:"insuranceSalary" json:"insuranceSalary"`
	InsuranceEmployee   float64 `bson:"insuranceEmployee" json:"insuranceEmployee"`
	InsuranceCompany    float64 `bson:"insuranceCompany" json:"insuranceCompany"`

	GrossSalary              float64 `bson:"grossSalary" json:"grossSalary"`
	TaxableIncome            float64 `bson:"taxableIncome" json:"taxableIncome"`
	PersonalExemptionApplied float64 `bson:"personalExemptionApplied" json:"personalExemptionApplied"`
	MedicalExemptionApplied  float64 `bson:"medicalExemptionApplied" json:"medicalExemptionApplied"`
	
	TaxAmount           float64 `bson:"taxAmount" json:"taxAmount"`
	MartyrsFund         float64 `bson:"martyrsFund" json:"martyrsFund"`
	
	NetSalary           float64 `bson:"netSalary" json:"netSalary"`
	
	Details             map[string]interface{} `bson:"details,omitempty" json:"details,omitempty"`
}